homeassistant/components/ambient_network/sensor.py

```diff
@@ -0,0 +1,332 @@
+    ),
+    SensorEntityDescription(
+        key=TYPE_WINDSPEEDMPH,
+        translation_key="wind_speed",
+        native_unit_of_measurement=UnitOfSpeed.MILES_PER_HOUR,
+        device_class=SensorDeviceClass.WIND_SPEED,
+        state_class=SensorStateClass.MEASUREMENT,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key=TYPE_YEARLYRAININ,
+        translation_key="yearly_rain",
+        native_unit_of_measurement=UnitOfPrecipitationDepth.INCHES,
+        device_class=SensorDeviceClass.PRECIPITATION,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        suggested_display_precision=2,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Ambient Network sensor platform."""
+
+    coordinator: AmbientNetworkDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    if coordinator.config_entry is not None:
+        async_add_entities(
+            AmbientNetworkSensor(
+                coordinator,
+                description,
+                coordinator.config_entry.data[ENTITY_STATION_NAME],
+            )
+            for description in SENSOR_DESCRIPTIONS
+            if AmbientNetworkSensor.get_sensor_value(coordinator.data, description.key)
+            is not None
+        )
+
+
+class AmbientNetworkSensor(AmbientNetworkEntity, SensorEntity):
+    """A sensor implementation for an Ambient Weather Network sensor."""
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: SensorEntityDescription,
+        station_name: str,
+    ) -> None:
+        """Initialize a sensor object."""
+
+        super().__init__(coordinator, description, station_name)
+        self._attr_suggested_display_precision = description.suggested_display_precision
+
+    def _update_attrs(self) -> None:
+        """Update sensor attributes."""
+
+        value = AmbientNetworkSensor.get_sensor_value(
+            self.coordinator.data, self.entity_description.key
+        )
+
+        # Treatments for special units.
+        if value is not None and self.device_class == SensorDeviceClass.TIMESTAMP:
+            value = datetime.fromtimestamp(value / 1000, tz=dt_util.DEFAULT_TIME_ZONE)
+
+        self._attr_native_value = value
+
+    @staticmethod
+    def get_sensor_value(station_data: dict[str, Any], sensor_key: str) -> Any | None:
+        """Return the sensor value from a station if available.
+
+        Eliminates data from stations that haven't been updated for a while.
+        """
+
+        if (created_at := station_data.get("created_at")) is None:
+            return None  # pragma: no cover
+
+        # Eliminate data that has been generated more than an hour ago. The station is
+        # probably offline.
+        if int(created_at / 1000) < int(
+            (datetime.now() - timedelta(hours=1)).timestamp()
+        ):
```

Comment:
Isn't this something for the coordinator?

Reply:
Good point. Moved to the coordinator.